<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ClientUI</class>
 <widget class="QWidget" name="ClientUI">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1260</width>
    <height>690</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>装配产线客户端</string>
  </property>
  <widget class="QGroupBox" name="group_box_device">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>20</y>
     <width>220</width>
     <height>135</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QGroupBox {
border-color: rgb(150, 150, 150);
border-width: 1px;
border-style: solid;
border-radius: 5px;
margin-top: 6px;
}
QGroupBox::title {
subcontrol-origin: margin;
subcontrol-position: top left;
left: 15px;
margin-left: 3px;
}</string>
   </property>
   <property name="title">
    <string/>
   </property>
   <widget class="QPushButton" name="button_scan">
    <property name="geometry">
     <rect>
      <x>15</x>
      <y>95</y>
      <width>85</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QComboBox" name="combo_box_device">
    <property name="geometry">
     <rect>
      <x>15</x>
      <y>55</y>
      <width>190</width>
      <height>25</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QPushButton" name="button_sub">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>95</y>
      <width>85</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QRadioButton" name="radio_camera">
    <property name="geometry">
     <rect>
      <x>15</x>
      <y>25</y>
      <width>85</width>
      <height>25</height>
     </rect>
    </property>
   </widget>
   <widget class="QRadioButton" name="radio_light">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>25</y>
      <width>85</width>
      <height>25</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_sub_msg">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>80</y>
      <width>85</width>
      <height>15</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_scan_msg">
    <property name="geometry">
     <rect>
      <x>15</x>
      <y>80</y>
      <width>85</width>
      <height>15</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
  </widget>
  <widget class="QGroupBox" name="group_box_status">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>175</y>
     <width>220</width>
     <height>215</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QGroupBox {
border-color: rgb(150, 150, 150);
border-width: 1px;
border-style: solid;
border-radius: 5px;
margin-top: 6px;
}
QGroupBox::title {
subcontrol-origin: margin;
subcontrol-position: top left;
left: 15px;
margin-left: 3px;
}</string>
   </property>
   <property name="title">
    <string/>
   </property>
   <widget class="QLabel" name="label_device_name">
    <property name="geometry">
     <rect>
      <x>15</x>
      <y>115</y>
      <width>190</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_device_index">
    <property name="geometry">
     <rect>
      <x>15</x>
      <y>25</y>
      <width>190</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLineEdit" name="edit_device_index">
    <property name="geometry">
     <rect>
      <x>15</x>
      <y>45</y>
      <width>190</width>
      <height>20</height>
     </rect>
    </property>
    <property name="alignment">
     <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLineEdit" name="edit_device_name">
    <property name="geometry">
     <rect>
      <x>15</x>
      <y>135</y>
      <width>190</width>
      <height>20</height>
     </rect>
    </property>
    <property name="alignment">
     <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLineEdit" name="edit_device_ip">
    <property name="geometry">
     <rect>
      <x>15</x>
      <y>90</y>
      <width>190</width>
      <height>20</height>
     </rect>
    </property>
    <property name="alignment">
     <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_device_ip">
    <property name="geometry">
     <rect>
      <x>15</x>
      <y>70</y>
      <width>190</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLineEdit" name="edit_status_message">
    <property name="geometry">
     <rect>
      <x>15</x>
      <y>180</y>
      <width>190</width>
      <height>20</height>
     </rect>
    </property>
    <property name="alignment">
     <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_status_message">
    <property name="geometry">
     <rect>
      <x>15</x>
      <y>160</y>
      <width>190</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
  </widget>
  <widget class="QPushButton" name="button_disconnect">
   <property name="geometry">
    <rect>
     <x>950</x>
     <y>605</y>
     <width>200</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QGroupBox" name="group_box_log">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>410</y>
     <width>220</width>
     <height>260</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QGroupBox {
border-color: rgb(150, 150, 150);
border-width: 1px;
border-style: solid;
border-radius: 5px;
margin-top: 6px;
}
QGroupBox::title {
subcontrol-origin: margin;
subcontrol-position: top left;
left: 15px;
margin-left: 3px;
}</string>
   </property>
   <widget class="QTextBrowser" name="browser_log">
    <property name="geometry">
     <rect>
      <x>15</x>
      <y>25</y>
      <width>190</width>
      <height>180</height>
     </rect>
    </property>
    <property name="html">
     <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Microsoft YaHei UI'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QPushButton" name="button_log_clear">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>220</y>
      <width>85</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="button_log_export">
    <property name="geometry">
     <rect>
      <x>15</x>
      <y>220</y>
      <width>85</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
  </widget>
  <widget class="QGroupBox" name="group_box_image">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>260</x>
     <y>20</y>
     <width>670</width>
     <height>650</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QGroupBox {
border-color: rgb(150, 150, 150);
border-width: 1px;
border-style: solid;
border-radius: 5px;
margin-top: 6px;
}
QGroupBox::title {
subcontrol-origin: margin;
subcontrol-position: top left;
left: 15px;
margin-left: 3px;
}</string>
   </property>
   <widget class="QPushButton" name="button_image_export">
    <property name="geometry">
     <rect>
      <x>15</x>
      <y>565</y>
      <width>150</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="button_image_clear">
    <property name="geometry">
     <rect>
      <x>15</x>
      <y>610</y>
      <width>150</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_image">
    <property name="geometry">
     <rect>
      <x>15</x>
      <y>25</y>
      <width>640</width>
      <height>480</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <bold>true</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(255, 255, 255);</string>
    </property>
    <property name="text">
     <string>No Image</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="button_image_load">
    <property name="geometry">
     <rect>
      <x>15</x>
      <y>520</y>
      <width>150</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="GainSpinSlider" name="widget_gain" native="true">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>520</y>
      <width>60</width>
      <height>105</height>
     </rect>
    </property>
   </widget>
   <widget class="GainSpinSlider" name="widget_r_gain" native="true">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>520</y>
      <width>60</width>
      <height>105</height>
     </rect>
    </property>
   </widget>
   <widget class="GainSpinSlider" name="widget_g_gain" native="true">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>520</y>
      <width>60</width>
      <height>105</height>
     </rect>
    </property>
   </widget>
   <widget class="GainSpinSlider" name="widget_b_gain" native="true">
    <property name="geometry">
     <rect>
      <x>590</x>
      <y>520</y>
      <width>60</width>
      <height>105</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_gain">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>625</y>
      <width>60</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>数字增益</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_r_gain">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>625</y>
      <width>60</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>红色增益</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_g_gain">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>625</y>
      <width>60</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>绿色增益</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_b_gain">
    <property name="geometry">
     <rect>
      <x>590</x>
      <y>625</y>
      <width>60</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>蓝色增益</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="Line" name="line_1">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>520</y>
      <width>30</width>
      <height>105</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>520</y>
      <width>30</width>
      <height>105</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="line_3">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>520</y>
      <width>30</width>
      <height>105</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="line_4">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>520</y>
      <width>30</width>
      <height>105</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="ExposureSpinSlider" name="widget_exposure" native="true">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>520</y>
      <width>60</width>
      <height>105</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_gain_2">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>625</y>
      <width>60</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>曝光度</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
  </widget>
  <widget class="QPushButton" name="button_exit">
   <property name="geometry">
    <rect>
     <x>950</x>
     <y>645</y>
     <width>200</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QGroupBox" name="group_box_light">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>950</x>
     <y>20</y>
     <width>290</width>
     <height>570</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QGroupBox {
border-color: rgb(150, 150, 150);
border-width: 1px;
border-style: solid;
border-radius: 5px;
margin-top: 6px;
}
QGroupBox::title {
subcontrol-origin: margin;
subcontrol-position: top left;
left: 15px;
margin-left: 3px;
}</string>
   </property>
   <property name="title">
    <string/>
   </property>
   <widget class="LightSpinSlider" name="widget_light_ch1" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>50</y>
      <width>60</width>
      <height>220</height>
     </rect>
    </property>
   </widget>
   <widget class="LightSpinSlider" name="widget_light_ch2" native="true">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>50</y>
      <width>60</width>
      <height>220</height>
     </rect>
    </property>
   </widget>
   <widget class="LightSpinSlider" name="widget_light_ch3" native="true">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>50</y>
      <width>60</width>
      <height>220</height>
     </rect>
    </property>
   </widget>
   <widget class="LightSpinSlider" name="widget_light_ch4" native="true">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>50</y>
      <width>60</width>
      <height>220</height>
     </rect>
    </property>
   </widget>
   <widget class="DelaySpinSlider" name="widget_delay_ch1" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>320</y>
      <width>60</width>
      <height>220</height>
     </rect>
    </property>
   </widget>
   <widget class="DelaySpinSlider" name="widget_delay_ch2" native="true">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>320</y>
      <width>60</width>
      <height>220</height>
     </rect>
    </property>
   </widget>
   <widget class="DelaySpinSlider" name="widget_delay_ch3" native="true">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>320</y>
      <width>60</width>
      <height>220</height>
     </rect>
    </property>
   </widget>
   <widget class="DelaySpinSlider" name="widget_delay_ch4" native="true">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>320</y>
      <width>60</width>
      <height>220</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_ch0">
    <property name="geometry">
     <rect>
      <x>12</x>
      <y>25</y>
      <width>50</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Ch1</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_ch2">
    <property name="geometry">
     <rect>
      <x>82</x>
      <y>25</y>
      <width>50</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Ch2</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_ch3">
    <property name="geometry">
     <rect>
      <x>152</x>
      <y>25</y>
      <width>50</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Ch3</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_ch4">
    <property name="geometry">
     <rect>
      <x>222</x>
      <y>25</y>
      <width>50</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Ch4</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_light">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>270</y>
      <width>270</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>————   光源亮度   ————</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_delay">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>540</y>
      <width>270</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>————   触发延时   ————</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>LightSpinSlider</class>
   <extends>QWidget</extends>
   <header location="global">light_spin_slider.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>DelaySpinSlider</class>
   <extends>QWidget</extends>
   <header location="global">delay_spin_slider.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>GainSpinSlider</class>
   <extends>QWidget</extends>
   <header location="global">gain_spin_slider.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ExposureSpinSlider</class>
   <extends>QWidget</extends>
   <header location="global">exposure_spin_slider.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>button_log_clear</sender>
   <signal>clicked()</signal>
   <receiver>browser_log</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>179</x>
     <y>552</y>
    </hint>
    <hint type="destinationlabel">
     <x>129</x>
     <y>474</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_exit</sender>
   <signal>clicked()</signal>
   <receiver>ClientUI</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>94</x>
     <y>689</y>
    </hint>
    <hint type="destinationlabel">
     <x>554</x>
     <y>362</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
